extra/release_info_generator.py

```diff
@@ -0,0 +1,131 @@
+from argparse import Namespace
+from calendar import isleap, monthrange
+from collections import namedtuple
+from contextlib import redirect_stdout
+from datetime import datetime, time, timedelta
+from io import StringIO
+from os import getenv
+from re import compile
+import json
+
+from release_tool import do_list_repos
+
+Tag = namedtuple("Tag", ["version", "datetime"])
+LtsEnd = namedtuple("LtsEnd", ["isExpired", "date"])
+
+releaseMatcher = compile(r"([0-9]+)\.([0-9]+)\.([0-9]+)")
+tagMatcher = compile(r"tag:'(.*)' datetime:'(\d+)'")
+
+lastChecked = getenv("LAST_CHECKED", "2.2.0")
+endOfToday = datetime.combine(datetime.now(), time.max)
+
+
+def determine_lts_end(tag, currentMinor, releases):
+    match = releaseMatcher.match(tag.version)
+    # for now we assume all even releases are LTS
+    if int(match.group(2)) % 2:
+        return
```

Comment:
This is already false because we are making 3.3 into an LTS. I don't really have a good solution for this other than manually adjusting it in the map, and then using it here as input... Can we pre-populate it with `lts` versions that don't exist yet, or will that throw the logic off?

Reply:
For now I put these checks in one place to be able to adjust the checks depending on the minor version... that should always work for as long as we keep the LTS schedule and won't require us to modify the versions file or LTS list manually, but it's still not super pretty...

Reply:
Ok, better than nothing I guess. The comment should be adjusted though.